pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -205,6 +225,26 @@ func (s *Server) watchRootCertAndGenKeyCert(stop <-chan struct{}) {
 	}
 }
 
+func (s *Server) watchDNSCertForK8sCA(pollingInterval time.Duration, stop <-chan struct{}, defaultCACertPath string, signerName string, approveCsr bool, requestedLifetime time.Duration) {
+	certUtil := certutil.NewCertUtil(int(defaultCertGracePeriodRatio * 100))
+	for {
+		if !sleep.Until(stop, pollingInterval) {
```

Comment:
why poll instead of using the waitTime?

Reply:
Using waitTime will restrict us to only one cert monitoring(if we want to extend this function to poll more than one cert). In WaitTime also we will need to sleep for 'waitTime' duration.
Also in other implementations in istio I have seen checking in particular interval for CA cert monitoring.

Reply:
We don't need to poll more than one cert ever. If we are just trying to renew a cert before it expires seems like we can do things a lot simpler

Reply:
Calculating waitTime and sleeping for waitTime before renewing the cert.